File: RoghtiaSystem_JahidKawa.Server/Data/Models/Medication.cs
namespace RoghtiaSystem_JahidKawa.Server.Data.Models
{
    public class Medication
    {
        public int Id { get; set; }
        public string Name { get; set; } = string.Empty;
        public string Type { get; set; } = string.Empty;
        public int Quantity { get; set; }
        public string Remarks { get; set; } = string.Empty;
        public int UserId { get; set; }
        public Users User { get; set; } = null!;
    }
}

File: roghtiasystem_jahidkawa.client/shared/types/medication.ts
export interface MedicationInput {
  name: string
  type: string
  quantity: number
  remarks: string
}

export interface Medication extends MedicationInput {
  id: number
  user: { id: number, userName: string }
}

export interface MedicationPage {
  items: Medication[]
  total: number
  page: number
  pageSize: number
}

export interface MedicationQuery {
  page?: number
  pageSize?: number
  search?: string
}
